src/test/components/ServiceWorkerManager.test.js

```diff
@@ -53,18 +47,53 @@ describe('app/ServiceWorkerManager', () => {
       writable: true,
       configurable: true,
     });
+
+    // We don't need a fullfledged service worker object because we're mocking
+    // the workbox library. But we need _something_ so that the app will try to
+    // add a service worker in the first place.
+    (navigator: any).serviceWorker = {};
   });
 
   afterEach(() => {
     process.env.NODE_ENV = 'development';
 
     Object.defineProperty(window, 'location', nativeLocation);
     nativeLocation = null;
+
+    delete navigator.serviceWorker;
   });
 
   function setup() {
+    // Comment out this spy when you want to debug this test file.
     jest.spyOn(console, 'log').mockImplementation(() => {});
 
+    // Due to the following jest issues, we can't put this implementation in a
+    // jest.mock() call and also spying on the Workbox constructor:
+    // https://github.com/facebook/jest/issues/7573
+    // https://github.com/facebook/jest/issues/10419
+    // But we need the constructor spy to access the instance.
```

Comment:
Thanks for the explanation and the issue links.
(also, ugh I hate the stale bot so much)